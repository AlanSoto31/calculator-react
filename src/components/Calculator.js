import React, { useState } from 'react';
import ButtonPanel from './ButtonPanel';
import Display from './Display';
// import Navbar from './Navbar';
import calculate from '../logic/calculate';
import '../CSS/styles.css';

const Calculator = () => {
  const [values, setValues] = useState({
    total: '0',
    next: null,
    operation: null,
  });

  const handleClick = (buttonName) => {
    const op = calculate(values, buttonName);
    setValues({
      total: op.total,
      next: op.next,
      operation: op.operation,
    });
  };

  return (
    <>
      <div className="calc-cont">
        <Display result={values.operation ? values.next : values.total} />
        <ButtonPanel clickHandler1={(buttonName) => handleClick(buttonName)} />
      </div>
    </>
  );
};

Calculator.defaultProps = { result: '0' };

export default Calculator;
