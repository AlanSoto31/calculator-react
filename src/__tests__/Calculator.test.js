import '@testing-library/jest-dom/extend-expect';
import Calculator from '../components/Calculator';
import renderer from 'react-test-renderer'


test("Calculator component renders", () => {
    const calculator = renderer.create(<Calculator/>).toJSON();
    expect(calculator).toMatchSnapshot();
});
