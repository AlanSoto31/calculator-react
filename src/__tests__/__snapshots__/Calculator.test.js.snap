// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders 1`] = `
<div
  className="calc-cont"
>
  <div
    className="disp-cont flex-e-c"
  >
    <p>
      0
    </p>
  </div>
  <div
    className="flex-c-c"
  >
    <div
      className="row-cont group-1 flex-sb-c"
    >
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0 ",
            "width": "25%",
          }
        }
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0 ",
            "width": "25%",
          }
        }
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0 ",
            "width": "25%",
          }
        }
        type="button"
      >
        %
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "background": "#F5913E",
            "width": "25%",
          }
        }
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="row-cont group-2 flex-sb-c"
    >
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0 ",
            "width": "25%",
          }
        }
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0 ",
            "width": "25%",
          }
        }
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0 ",
            "width": "25%",
          }
        }
        type="button"
      >
        9
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "background": "#F5913E",
            "width": "25%",
          }
        }
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="row-cont group-3 flex-sb-c"
    >
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0 ",
            "width": "25%",
          }
        }
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0 ",
            "width": "25%",
          }
        }
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0 ",
            "width": "25%",
          }
        }
        type="button"
      >
        6
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "background": "#F5913E",
            "width": "25%",
          }
        }
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="row-cont group-4 flex-sb-c"
    >
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0 ",
            "width": "25%",
          }
        }
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0 ",
            "width": "25%",
          }
        }
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0 ",
            "width": "25%",
          }
        }
        type="button"
      >
        3
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "background": "#F5913E",
            "width": "25%",
          }
        }
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="row-cont group-5 flex-sb-c"
    >
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0 ",
            "width": "50%",
          }
        }
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0 ",
            "width": "25%",
          }
        }
        type="button"
      >
        .
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "background": "#F5913E",
            "width": "25%",
          }
        }
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
